app/helpers/blacklight/hierarchy_helper.rb

```diff
@@ -2,20 +2,21 @@ module Blacklight::HierarchyHelper
   # Putting bare HTML strings in a helper sucks. But in this case, with a
   # lot of recursive tree-walking going on, it's an order of magnitude faster
   # than either render(:partial) or content_tag
-  def render_facet_hierarchy_item(facet_item, facet_config)
+  def render_facet_hierarchy_item(facet_item, facet_config, level)
     field_name = facet_item.field
-    subset = facet_item.items
-    li_class = subset.empty? ? 'h-leaf' : 'h-node'
+    hierarchy_components = facet_config[:sub_hierarchy]
+    subset = level < hierarchy_components.size ? facet_item.subs&.[](hierarchy_components[level]) : nil
```

Comment:
it's hard for me to understand what is going on here

Reply:
Extracted this into a method and documented it; hopefully clearer now?